### DISCUSSION TITLE
Text with imaginary lines is being treated as a table

### DISCUSSION THREAD

**Discussion by sachinnethakanipersonal** [jsvine/pdfplumber#1139] [Ask for help with specific PDFs]
## Describe the bug

We are trying to extract tables out of pdf file and we are trying to export them as an image for our further analysis. But what we have found is there are few cases where we have found that the page.debug_tablefinder() is extracting the text more precisely paragraphs also with imaginary lines as a table

## Code to reproduce the problem
**_page.debug_tablefinder({"intersection_tolerance":8})_**

## Expected behavior

Only the true tables with proper borders or lines are extracted

## Actual behavior

Text with imaginary lines/borders is being treated as a table 

## Screenshots

![image](https://github.com/jsvine/pdfplumber/assets/127970790/da80b186-abb7-471c-a4c1-215386bf134a)
this is one of the examples which is treated as a table

## Environment

- pdfplumber version: 0.11.0
- Python version: 3.12.2
- OS: Windows/Databricks

## Additional context

Apologies, I may not be able to share the documents as they are part of the legal contracts

**Comment by jsvine** [comment]
Without access to the PDF itself, it will be difficult to provide suggestions, unfortunately. But you may find some [similar examples in the discussions by searching for "invisible lines"](https://github.com/jsvine/pdfplumber/discussions?discussions_q=%22invisible+lines%22).

**Comment by MycatQieZi** [comment]
In some PDFs I’ve come across, “rectangle” objects occasionally appear that are meaningless and useless—they have no fill color and no border line. Yet TableFinder still treats them as Edges and outputs their content as part of a table. As a temporary fix, I filter out rectangles that have no fill or are filled white and that also have no stroke color or a white stroke color, so that the body text can be parsed correctly. The drawback is that this makes the detection problem for truly borderless tables even worse.

**Comment by MycatQieZi** [comment]
You can customize the filtering logic in `utils/geometry.py`(specifically the `filter_edges` function) to tighten the criteria for edge removal and better handle these edge-case rectangles.

Hope this helps！
